api/nodes/filters.py

```diff
@@ -57,14 +60,59 @@ def convert_special_params_to_odm_query(self, field_name, query_params, key, dat
             if utils.is_falsy(query_params[key]):
                 # Use `or` when looking for not-preprints, to include both no file and is_orphaned
                 sub_query = functools.reduce(operator.or_, [
-                    Q(item['source_field_name'], item['op'], item['value'])
+                    MQ(item['source_field_name'], item['op'], item['value'])
                     for item in data
                 ])
             else:
                 sub_query = functools.reduce(operator.and_, [
-                    Q(item['source_field_name'], item['op'], item['value'])
+                    MQ(item['source_field_name'], item['op'], item['value'])
                     for item in data
                 ])
             return sub_query
         else:
             raise InvalidFilterError('Expected type list for field {}, got {}'.format(field_name, type(data)))
+
+
+class NodesFilterMixin(ListFilterMixin):
+
+    def postprocess_query_param(self, key, field_name, operation):
+        pass
+
+    def filter_by_field(self, queryset, field_name, operation):
+        if field_name == 'parent':
+            if operation['op'] == 'eq':
+                if operation['value']:
+                    # filter[parent]=<nid>
+                    parent = utils.get_object_or_error(Node, operation['value'], display_name='parent')
+                    return queryset.filter(guids___id__in=[node._id for node in parent.get_nodes(is_node_link=False)])
+                # else filter[parent]=null
+                return queryset.get_roots()
+            elif operation['op'] == 'ne':
+                if not operation['value']:
+                    # filter[parent][ne]=null
+                    child_ids = [each.child.id for each in (
+                        NodeRelation.objects.filter(is_node_link=False, child___contributors=self.get_user())
+                        .exclude(parent__type='osf.collection')
+                        .exclude(child__is_deleted=True)
+                    )]
+                    return queryset.filter(id__in=set(child_ids))
+                # else filter[parent][ne]=<nid>
+                raise InvalidFilterValue()
+            else:
+                # filter[parent][gte]=''
+                raise InvalidFilterOperator()
```

Comment:
Pass `value` and `value_operators` here so the error message will be more useful.